src/source_sender/mod.rs

```diff
@@ -322,7 +379,12 @@ impl Inner {
     }
 
     async fn send_event(&mut self, event: impl Into<EventArray>) -> Result<(), ClosedError> {
-        self.send(event.into()).await
+        let event: EventArray = event.into();
+        let count = event.len();
+        let mut unsent_event_count = UnsentEventCount::new(count);
+        let res = self.send(event).await;
+        unsent_event_count.discard();
+        res
```

Comment:
This, on the surface, looks like `UnsentEventCount::drop` would never happen with `count > 0`. Is this meant to handle where the task is dropped without sending the events? If so, it might be worth a comment to that effect.

Reply:
That's correct. This is to account for the scenarios where the task tis dropped on line 385. I can add a comment.